DEVX-2354: Update demo to use FMC kafka-connect-datagen: beginner CCloud CLI (#887)
confluentinc/examples: ccloud/docs/beginner-cloud.rst

-  `Run a Java producer without ACLs`_
-  `Run a Java producer with ACLs`_
-  `Run a Java producer with a prefixed ACL`_

> -  `Run a fully managed datagen_ccloud_pageviews connector`_

Reviewer: ```suggestion
-  `Run a fully managed Confluent Cloud connector`_
```

   You should see two ``Deleted ACLs.`` messages.

> Run a fully managed datagen_ccloud_pageviews connector

Reviewer: ```suggestion
Run a fully managed Confluent Cloud connector
```

         ServiceAccountId | Permission | Operation | Resource |  Name   |  Type
       +------------------+------------+-----------+----------+---------+---------+
         User:104349      | ALLOW      | WRITE     | TOPIC    | *       | LITERAL

#. Create a local configuration file
   :devx-examples:`datagen_ccloud_pageviews.json|ccloud/beginner-cloud/datagen_ccloud_pageviews.json`
   with |ccloud| connection information. Substitute your API key and secret for the service account,
   in the ``kafka.api.key`` and ``kafka.api.secret`` fields. See below for an example:

   .. literalinclude:: ../beginner-cloud/datagen_ccloud_pageviews.json

#. Create a managed connector in Confluent Cloud with the configuration file you made in the
   previous step using the following commands:

   .. code-block:: text

      ccloud connector create --config datagen_ccloud_pageviews.json

   Your output should resemble:

   .. code-block:: text

      Created connector datagen_ccloud_pageviews lcc-zno83

> #. The connector may take up to 5 minutes to provision, use the following command to check if the

Reviewer: ```suggestion
#. The connector may take up to 5 minutes to provision. Run the following command to check the connector status
```

>    connector is `PROVISIONING` or `RUNNING`.

Reviewer: ```suggestion
```

   .. code-block:: bash

      ccloud connector list

   Your output should resemble the following:

   .. code-block:: text

           ID     |           Name            |    Status    |  Type  | Trace
      +-----------+---------------------------+--------------+--------+-------+
        lcc-zno83 | datagen_ccloud_pageviews  | PROVISIONING | source |

>    If the ``Status`` is ``RUNNING`` you may move on to the next step.

Reviewer: ```suggestion
   When the ``Status`` is ``RUNNING`` you may move on to the next step.
```

      ../../ccloud/ccloud-generate-cp-configs.sh /tmp/client.config &>/dev/null
      source delta_configs/env.delta

#. Review the managed connector config file :devx-examples:`datagen_ccloud_pageviews.json|ccloud/beginner-cloud/datagen_ccloud_pageviews.json`:

   .. literalinclude:: ../beginner-cloud/datagen_ccloud_pageviews.json

#. Create a managed connector in Confluent Cloud with the provided configurations mentioned above using the following commands:

   .. code-block:: text

      source ../../utils/ccloud_library.sh 

>       ccloud::create_connector datagen_ccloud_pageviews.json

Reviewer: This is a library function which is great for automation (and the `start.sh` script) but for the docs it is more instructive to have the user learn the Confluent Cloud CLI command `ccloud connector create` instead.

      Created connector datagen_ccloud_pageviews lcc-zno83

#. Run the following command to verify connector is running:

   .. code-block:: bash

>       ccloud::wait_for_connector_up datagen_ccloud_pageviews.json 300

Reviewer: This is a library function which is great for automation (and the `start.sh` script) but for the docs it is more instructive to have the user learn the Confluent Cloud CLI command `ccloud connector <list>` (or `<describe>`) instead.

      Created connector datagen_ccloud_pageviews lcc-zno83

#. Run the following command to verify connector is running:

   .. code-block:: bash

>       ccloud::wait_for_connector_up connectors/ccloud-datagen-pageviews.json 240 || exit 1

Reviewer 2: I'm not sure we'd want to have the `exit` conditional in the directions. If we do the user will lose their shell instance if the command doesn't succeed

Author: I'm glad you pointed that out. I removed that bit in the instructions and corrected the path to the .json config in this [bc73545](https://github.com/confluentinc/examples/pull/887/commits/bc73545dabaa3e4b8b011366fd24b25b0ee43d22)

   .. code-block:: bash

      ccloud kafka acl delete --allow --service-account 104349 --operation WRITE --topic '*'

   You should see a ``Deleted ACLs.`` message after running each of the previous
   commands.

Clean up your Confluent Cloud resources
---------------------------------------
#. Complete the following steps to delete the managed connector:

   a. Find the connector ID:
      
      .. code-block:: bash

         ccloud connector list
	 
      Which should display a something similar to below. Locate your connector ID, in this case the connector ID is ``lcc-zno83``.

      .. code-block:: text

              ID     |           Name           | Status  |  Type  | Trace
         +-----------+--------------------------+---------+--------+-------+
           lcc-zno83 | datagen_ccloud_pageviews | RUNNING | source |

>    b. Delete the connector using the ID:

Reviewer: ```suggestion
   b. Delete the connector, referencing the connector ID from the previous step:
```
